src/averageAges.js

```diff
@@ -37,7 +40,20 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filterWom = people.filter(wom => withChildren
+    ? findMomByName(people, wom) : wom.sex === 'f');
+
+  const ageWom = filterWom.map((age) => differenceAge(age));
+
+  return calculatePeoplesAge(ageWom);
+}
+
+function findMomByName(people, wom) {
+  return people.some(nameMother => wom.name === nameMother.mother);
+}
+
+function differenceAge(age) {
```

Comment:
Similar to previous comments, consider renaming `age` to `person` for better readability.